### PR TITLE
Note taking

### PR DESCRIPTION
**Pull request by bhtibrewal**
Added note-taking to each video 
Created custom Controller

Pages to check
- NotesController.js
- NotesForm.js
- NotesSection.js
- removeAllNotesForVideo.js
- addNewNoteToVideo.js
- deleteNotefromVideo.js

### CODE REVIEW

**Review by Naman-Saxena1 — CHANGES_REQUESTED** [review]
Code looks fine 
But I have 2 suggestions:
1. Clear Notes button is not working, you can cross check
2. Contrast ratio of notes is too low 
Screenshot (1)

File: src/backend/controllers/AuthController.js [hunk 1]
```diff
@@ -38,6 +38,8 @@ export const signupHandler = function (schema, request) {
       likes: [],
       history: [],
       playlists: [],
+      watchlater: [],
+      notesManagement: [],
```
Comment (Ankur9669) [inline src/backend/controllers/AuthController.js:42]: Appreciation: Nice work on the controller part

File: src/pages/emptylist_page/EmptyListPage.jsx [hunk 2]
```diff
@@ -1,17 +1,21 @@
...
         <p className="body-l">{text}</p>
-        <ButtonPrimary onClick={() => navigate("/videos")}>
-          <span> Explore</span>
-          <i className="fa-solid fa-arrow-right"></i>
-        </ButtonPrimary>
+        {children ? (
+          children
+        ) : (
+          <ButtonPrimary onClick={() => navigate("/videos")}>
+            <span> Explore</span>
```
Comment (Ankur9669) [inline src/pages/emptylist_page/EmptyListPage.jsx:15]: Could use p tag here

File: src/pages/video_page/video_page.css [hunk 3]
```diff
@@ -0,0 +1,21 @@
+.notes-management {
+  max-width: 800px;
```
Comment (Ankur9669) [inline src/pages/video_page/video_page.css:2]: can use relative units

File: src/server.js [hunk 4]
```diff
@@ -131,6 +140,13 @@ export function makeServer({ environment = "development" } = {}) {
...
+
+      /* notes api (private) */
+      this.delete("/user/notes/:videoId", removeAllNotesForVideoHandler.bind(this));
+      this.get("/user/notes/:videoId", getAllNotesForVideoHandler.bind(this));
+      this.post("/user/notes/:videoId", addNewNoteToVideoHandler.bind(this));
+      this.post("/user/notes/:videoId/:noteId", updateNoteHandler.bind(this));
+      this.delete("/user/notes/:videoId/:noteId", deleteNoteFromVideoHandler.bind(this));
     },
   });
 }
```
Comment (Ankur9669) [inline src/server.js:152]: Nice work here

**Review by Ankur9669 — APPROVED** [review]
The code and the UI both are looking good,

image

On clicking clear notes button the notes ae not getting cleared

You can correct that than you can merge

**Pull request merged**